### ISSUE TITLE
Brainf**k

### ISSUE THREAD

**Issue by heymrhayes** [adambard/learnxinyminutes-docs#2120]
I teach CS in high school and don't feel comfortable pointing my students (some as young as 13) to the page when Brainf**k is, unfortunately, the first thing on the page.

Certainly the word is not new to my students (although certainly not all use it and some are certainly offended by it), but I don't want to be shoving it in their faces.

Maybe we can mask the spelling?

**Comment by danilolekovic** [comment]
Perhaps it should be changed to its file extension, 'bf'?

**Comment by ryanmavilia** [comment]
I like the idea of changing it to 'bf' especially since if we changed it to brainf**k like a lot of sites do it would still be recognized. This actually came up in my uni's lecture when the teacher visited repl.it laughs were had since we are much older, but for high school I can see a very big issue here.

**Comment by heymrhayes** [comment]
+1 for the "bf" option.

**Comment by davelima** [comment]
bf :+1:

**Comment by JacobCWard** [comment]
+1 for 'bf'

**Comment by adambard** [comment]
Good point, it is done.

**Comment by ian-bertolacci** [comment]
When did it get changed back?

**Comment by prertik** [comment]
Looks like some problems occured here. The filename and language was changed to pass the tests. So, what should the naming be? I'm open for discussions and change. @adambard

**Comment by ian-bertolacci** [comment]
Im not sure how the testing works, and the errors shown in #2832 are some-what ambiguous.
But I think the error "filename missing" means that the `filename` field in the language's markdown file is missing.
Previous to 096fcc97902ba17b6a8bceea4492757b49ca864a there was no `filename` field, which would have caused this test to fail.

I think if we simply rename BrainFuck to bf (or some other variant) in all cases everything should work.
But that's a guess based on my limited interpretation of what the test does.

**Comment by prertik** [comment]
You are correct. Looking back, I found the filename was missing. Yes, i think there wasn't any problem with the naming field.

**Comment by jc260552** [comment]
Should Brainfuck also be changed on the front page?

**Comment by ian-bertolacci** [comment]
Yes.

**Comment by thornjad** [comment]
Brainfuck currently displays as bf on the site, is there anything left to do before closing this issue?

**Comment by ian-bertolacci** [comment]
I believe this issue can be closed.
